### ISSUE TITLE
Issues with installation one-liners

### ISSUE THREAD

**Issue by cdrage** [redhat-developer/odo#523]
I'm having issues with the Curl one-liners: 

```sh
▶ sudo curl -L  "https://github.com/redhat-developer/odo/releases/download/v0.0.6/odo-linux-amd64.gz" | gzip -d > /usr/local/bin/odo; chmod +x /usr/local/bin/odo

zsh: permission denied: /usr/local/bin/odo
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   609    0   609    0     0   2009      0 --:--:-- --:--:-- --:--:--  2009
  0 7759k    0 16360    0     0  28802      0  0:04:35 --:--:--  0:04:35 80990
curl: (23) Failed writing body (0 != 16360)
chmod: cannot access '/usr/local/bin/odo': No such file or directory
```

I've removed `/usr/local/bin/odo` (old odo).

I've google'd around and found people with similar issues (https://github.com/kubernetes/helm/issues/2802)

Why do we have to download tarballs and worry about using gzip? Can we just download the binaries directly from GitHub?

**Comment by kadel** [comment]
For someone with slow internet, 28 MB vs 8 MB can mean a huge difference. 

The problem with the command is that the pipe is not executed as root, it is performed in the users shell.

We have to run the whole command in the same shell under sudo.
```sh
sudo sh -c 'curl -L  "https://github.com/redhat-developer/odo/releases/download/v0.0.6/odo-linux-amd64.gz" | gzip -d > /usr/local/bin/odo; chmod +x /usr/local/bin/odo'
```


Interesting is that on macOS original command works. But in Linux you have to wrap it whole in the shell as I mentioned.

**Comment by cdrage** [comment]
Could we instead include options for downloading either the binary or a tarball? @kadel

**Comment by surajnarwade** [comment]
Fixed in https://github.com/redhat-developer/odo/pull/526

**Comment by syamgk** [comment]
I think this can be closed since we merged #526
